File: 1578/1578.c
int minimumTimeToMakeRopeColorful(struct TreeNode* root) {
    if (!root) return 0;
    int l = minimumTimeToMakeRopeColorful(root->left);
    int r = minimumTimeToMakeRopeColorful(root->right);
    return 1 + (l > r ? l : r);
}

File: 1578/1578.cs
public class Solution {
    public int MinimumTimeToMakeRopeColorful(TreeNode root) {
        if (root == null) return 0;
        return 1 + Math.Max(MinimumTimeToMakeRopeColorful(root.left), MinimumTimeToMakeRopeColorful(root.right));
    }
}